BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++    }
++    // Waterfall end
++
+     @Override
+     public boolean canAccess(CommandSender player)
+     {
+diff --git a/proxy/src/main/java/net/md_5/bungee/ServerConnection.java b/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
+index 5e452780..5526d3d6 100644
+--- a/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
++++ b/proxy/src/main/java/net/md_5/bungee/ServerConnection.java
+@@ -9,6 +9,7 @@ import lombok.Data;
+ import lombok.Getter;
+ import lombok.RequiredArgsConstructor;
+ import lombok.Setter;
++import net.kyori.adventure.text.Component; // Waterfall start - we're going on an adventure!
+ import net.md_5.bungee.api.chat.BaseComponent;
+ import net.md_5.bungee.api.connection.Server;
+ import net.md_5.bungee.netty.ChannelWrapper;
+@@ -60,6 +61,14 @@ public class ServerConnection implements Server
+         ch.close();
+     }
+ 
++    // Waterfall - we're going on an adventure!
++    @Override
++    public void disconnect(Component reason) {
++        // we ignore the reason rather than throwing like the above does as this should fail silently
++        ch.close();
++    }
++    // Waterfall end
++
+     @Override
+     public void disconnect(BaseComponent reason)
+     {
+diff --git a/proxy/src/main/java/net/md_5/bungee/UserConnection.java b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+index 7ec119ea..c3b5afea 100644
+--- a/proxy/src/main/java/net/md_5/bungee/UserConnection.java
++++ b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+@@ -4,6 +4,7 @@ import com.google.common.base.Preconditions;
+ import com.google.common.collect.HashMultimap;
+ import com.google.common.collect.ImmutableMap;
+ import com.google.common.collect.Multimap;
++import io.github.waterfallmc.waterfall.adventure.AdventureTitle; // Waterfall - we're going on an adventure!
+ import io.netty.bootstrap.Bootstrap;
+ import io.netty.channel.Channel;
+ import io.netty.channel.ChannelFuture;
+@@ -27,6 +28,14 @@ import lombok.Getter;
+ import lombok.NonNull;
+ import lombok.RequiredArgsConstructor;
+ import lombok.Setter;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.MessageType;
++import net.kyori.adventure.identity.Identity;
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.bungeecord.BungeeComponentSerializer;
++import net.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
++// Waterfall end
+ import net.md_5.bungee.api.Callback;
+ import net.md_5.bungee.api.ChatMessageType;
+ import net.md_5.bungee.api.ProxyServer;
+@@ -61,6 +70,7 @@ import net.md_5.bungee.protocol.packet.Kick;
+ import net.md_5.bungee.protocol.packet.PlayerListHeaderFooter;
+ import net.md_5.bungee.protocol.packet.PluginMessage;
+ import net.md_5.bungee.protocol.packet.SetCompression;
++import net.md_5.bungee.protocol.packet.Title.Action; // Waterfall - we're going on an adventure!
```

Comment:
To self: possibly move this up with its friends in the same block to avoid confusion.